### PR TITLE
Doc: Overview of nrf5340_audio features

### PR DESCRIPTION
**Pull request by asbjornsabo**
Add an overview of nRF5340_audio application features to the documentation.
The purpose is to give users an overview of which features exist, and how mature they are, to set expectations correctly.

### CODE REVIEW

**Comment by asbjornsabo** [comment]
(Force-pushed to fix build error - there was one extra column in one of the rows in the table.)

**Comment by greg-fer** [comment]
@asbjornsabo , added commit with my review.

- I moved the table to the Software maturity levels page (discussed with @carlescufi  and @tejlmand ) and then included it from there on the application page.
- The table at the start of the page in a separate new section was hiding the two overview subsections and overlapping with the ToC, so I added an expandable field at the very start of the doc that includes the table.
- Edited the table contents, as with the original setup most of the acronyms were being expanded later in the docs and it has the be otherwise. I also tried to put the table after most of the acronyms are expanded and explained, but it would not work, hence the current solution.

**Review by carlescufi — APPROVED** [review]

**Review by greg-fer — APPROVED** [review]
Please squash my commit before merge.

**Comment by asbjornsabo** [comment]
Do not merge yet - will propose a minor change.

**Review by greg-fer — APPROVED** [review]

**Comment by asbjornsabo** [comment]
Added the last change, squashed all into one commit.

**Review by koffes — CHANGES_REQUESTED** [review]

File: doc/nrf/software_maturity.rst [hunk 1]
```diff
@@ -73,10 +73,89 @@ See the following table for more details:
...
+
+        - Experimental
+      * - **Broadcast sink**
+        - Receiving broadcast audio using BIS and BIG.
+
+          Synchronizes and unsynchronizes with the stream.
+        - The following limitations apply:
+
+          * BAP broadcast, one BIG, one of the two BIS streams (selectable).
+          * Audio output: I2S (Headset output).
```
[the remark this answers is not in the document]
Comment (asbjornsabo) [reply doc/nrf/software_maturity.rst]: Changed

File: doc/nrf/software_maturity.rst [hunk 2]
```diff
@@ -73,10 +73,89 @@ See the following table for more details:
...
+        - The following limitations apply:
+
+          * BAP unicast, one CIG, one of the two CIS streams (selectable).
+          * Audio output: I2S (Headset).
+          * Configuration: 48 kHz, 16 bit, several bit rates ranging from 24 kbps to 160 kbps.
+
+        - Experimental
+
+.. _software_maturity_protocol:
+
```
[the remark this answers is not in the document]
Comment (asbjornsabo) [reply doc/nrf/software_maturity.rst:154]: Discussed with @koffes :
Does not fit well - the other things listed are features, while the controller is a component.

**Review by rick1082 — APPROVED** [review]
Looks good to me.

File: applications/nrf5340_audio/README.rst [hunk 3]
```diff
@@ -13,6 +13,19 @@ It is developed for use with the :ref:`nrf53_audio_app_dk`.
 In its default configuration, the application requires the :ref:`LC3 software codec <nrfxlib:lc3>`.
 The application also comes with various tools, including the :file:`buildprog.py` Python script that simplifies building and programming the firmware.
 
+.. _nrf53_audio_app_overview_features:
+
+Feature support matrix
+   The following table lists features of the nRF5340 Audio application and their respective limitations and maturity level.
```
Comment (erikrobstad) [inline applications/nrf5340_audio/README.rst:19]: To me it looks strange that this text is "tabbed out"

File: doc/nrf/software_maturity.rst [hunk 4]
```diff
@@ -73,10 +73,89 @@ See the following table for more details:
...
+          * BAP unicast, one CIG with two CIS.
+          * Audio input: USB or I2S (Line in or using Pulse Density Modulation).
+          * Configuration: 48 kHz, 16 bit, several bit rates ranging from 24 kbps to 160 kbps.
+
+        - Experimental
+      * - **Unicast sink**
+        - BAP unicast, 1 CIG with 2 CIS streams.
+
+          Context type setting (for forcing enabling and disabling streaming, to emulate play and pause).
```
Comment (erikrobstad) [inline doc/nrf/software_maturity.rst:144]:
```suggestion
          Play and pause emulated by forcing enabling and disabling streaming by changing available contexts.
```

**Review by erikrobstad — APPROVED** [review]

File: doc/nrf/software_maturity.rst (hunk 4 above)
Comment (greg-fer) [reply doc/nrf/software_maturity.rst:144]: With your edit, it is unclear what is emulated by what due to double "by" structure.

File: applications/nrf5340_audio/README.rst (hunk 3 above)
Comment (greg-fer) [reply applications/nrf5340_audio/README.rst:19]: It is required to have the maturity table included at the very beginning of the doc and avoid breaking the structure of the page.

File: doc/nrf/software_maturity.rst (hunk 4 above)
Comment (greg-fer) [reply doc/nrf/software_maturity.rst:144]: Discussed over Teams with @erikrobstad :
```suggestion
          To emulate play and pause, the available context type for media is added or removed. This enables and disables streaming, respectively.
```

Comment (asbjornsabo) [reply doc/nrf/software_maturity.rst]: Changed as proposed.

File: doc/nrf/software_maturity.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (asbjornsabo) [reply doc/nrf/software_maturity.rst:154]: Agreed to keep as is.

**Review by koffes — APPROVED** [review]

**Pull request merged**